### ISSUE TITLE
Bug: `generate_if_with_label` doesn't catch `else` without `begin`/`end` delimiters.

### ISSUE THREAD

**Issue by DaveMcEwan** [dalance/svlint#207]
```systemverilog
module M;
  if (x) begin: l_abc
    assign a = 0;
  end else if (y) begin: l_def
    assign a = 1;
  end else
    assign a = 2;
endmodule
```

- Current behavior: Checks for label only when `begin`/`end` delimiters exist on else clause.
- Intended behavior: Same as if and if-else clauses, fail when delimiters don't exist.
